pkg/apis/sessions/session_state.go

```diff
@@ -235,35 +251,3 @@ func lz4Decompress(compressed []byte) ([]byte, error) {
 
 	return payload, nil
 }
-
-// validate ensures the decoded session is non-empty and contains valid data
-//
-// Non-empty check is needed due to ensure the non-authenticated AES-CFB
-// decryption doesn't result in garbage data that collides with a valid
-// MessagePack header bytes (which MessagePack will unmarshal to an empty
-// default SessionState). <1% chance, but observed with random test data.
-//
-// UTF-8 check ensures the strings are valid and not raw bytes overloaded
-// into Latin-1 encoding. The occurs when legacy unencrypted fields are
-// decrypted with AES-CFB which results in random bytes.
-func (s *SessionState) validate() error {
```

Comment:
Now that Locks & Clocks are in the Session for internal usage and not all fields are serialized, the `reflect.DeepEqual` might have issues moving forward (in earlier iterations, it did not like the `Clock` mutex).

This really isn't needed anymore and is an unneeded computation in the session deserialization that happens every request. I added an important note on the migration considerations for the slim user base this might affect.